File: rwrdc.py
# python code
def RWRDC(adj_matrix, c=0.05):
    n = adj_matrix.shape[0]
    degree_matrix = np.diag(np.sum(adj_matrix, axis=1))
    inv_degree_matrix = np.linalg.pinv(degree_matrix)
    w = inv_degree_matrix @ adj_matrix
    identity_matrix = np.eye(n)
    
    p = np.zeros((n, n))
    for k in range(n):
        r0 = np.zeros(n)
        r0[k] = 1
        p[:, k] = (1 - c) * np.linalg.pinv(identity_matrix - c * w.T) @ r0
    
    s = p + p.T
    return s

File: RWRDC.m
%MATLAB code
%input:adjacency matrix A
%output:efficacy score S
function S=RWRDC(A)
c=0.05;%restart probability
[cc,cc]=size(A);
w=inv(diag(sum(A,2)))*A;
for k=1:cc
p0=zeros(cc,1);
p0(k)=1; 
p1=p0;
p(:,k)=(1-c)*w'*p0+c*p0;
while max(abs(p(:,k)-p1))>10^(-6)
    p1=p(:,k);
    p(:,k)=(1-c)*w'*p1+c*p0;
end
end
S=p+p';
end
